packages/flutter/lib/src/painting/flutter_logo.dart

```diff
@@ -31,47 +31,43 @@ enum FlutterLogoStyle {
   stacked,
 }
 
-const int _lightShade = 400;
-const int _darkShade = 900;
-const Map<int, Color> _kDefaultSwatch = const <int, Color>{
-  _lightShade: const Color(0xFF42A5F5),
-  _darkShade: const Color(0xFF0D47A1)
-};
-
 /// An immutable description of how to paint Flutter's logo.
 class FlutterLogoDecoration extends Decoration {
   /// Creates a decoration that knows how to paint Flutter's logo.
   ///
-  /// The [swatch] controls the color used for the logo. The [style] controls
+  /// The [colors] control the colors used for the logo. The [style] controls
   /// whether and where to draw the "Flutter" label. If one is shown, the
   /// [textColor] controls the color of the label.
   ///
-  /// The [swatch], [textColor], and [style] arguments must not be null.
+  /// The [colors], [textColor], and [style] arguments must not be null.
   const FlutterLogoDecoration({
-    this.swatch: _kDefaultSwatch,
+    this.lightColor: const Color(0xFF42A5F5), // Colors.blue[400]
+    this.darkColor: const Color(0xFF0D47A1), // Colors.blue[900]
     this.textColor: const Color(0xFF616161),
     this.style: FlutterLogoStyle.markOnly,
     this.margin: EdgeInsets.zero,
   }) : _position = style == FlutterLogoStyle.markOnly ? 0.0 : style == FlutterLogoStyle.horizontal ? 1.0 : -1.0, // ignore: CONST_EVAL_TYPE_BOOL_NUM_STRING
        // (see https://github.com/dart-lang/sdk/issues/26980 for details about that ignore statement)
        _opacity = 1.0;
 
-  FlutterLogoDecoration._(this.swatch, this.textColor, this.style, this._position, this._opacity, this.margin);
+  FlutterLogoDecoration._(this.lightColor, this.darkColor, this.textColor, this.style, this._position, this._opacity, this.margin);
 
-  /// The colors to use to paint the logo. This map should contain at least two
-  /// values, one for 400 and one for 900.
+  /// The logo graphic is painted with [lightColor] and [darkColor].
   ///
-  /// If possible, the default should be used. It corresponds to the
-  /// [Colors.blue] swatch from the Material library.
+  /// If possible, the default should be used. It corresponds to the 400 and 900
+  /// values of [Colors.blue] from the Material library.
   ///
-  /// If for some reason that color scheme is impractical, the [Colors.amber],
-  /// [Colors.red], or [Colors.indigo] swatches can be used. These are Flutter's
-  /// secondary colors.
+  /// If for some reason that color scheme is impractical, the same entries from
+  /// [Colors.amber], [Colors.red], or [Colors.indigo] colors can be used. These
+  /// are Flutter's secondary colors.
   ///
   /// In extreme cases where none of those four color schemes will work,
-  /// [Colors.pink], [Colors.purple], or [Colors.cyan] swatches can be used.
+  /// [Colors.pink], [Colors.purple], or [Colors.cyan] can be used.
   /// These are Flutter's tertiary colors.
-  final Map<int, Color> swatch;
+  final Color lightColor;
+
+  /// See [lightColor].
```

Comment:
add something like:

```
/// The darker of the two colors used to paint the logo.
```

...before a blank line then `See [lightColor] for a discussion of which colors are reasonable` or some such.